packages/date-time/src/components/DatePicker/DatePicker.base.tsx

```diff
@@ -205,91 +243,263 @@ export class DatePickerBase extends React.Component<IDatePickerProps, IDatePicke
+      }
+    };
+
+    /**
+     * Callback for closing the calendar callout
+     */
+    const calendarDismissed = (newlySelectedDate?: Date): void => {
+      preventNextFocusOpeningPicker();
+      dismissDatePickerPopup(newlySelectedDate);
+      // don't need to focus the text box, if necessary the focusTrapZone will do it
+    };
+
+    const handleEscKey = (ev: React.KeyboardEvent<HTMLElement>): void => {
+      ev.stopPropagation();
+      calendarDismissed();
+    };
 
     const classNames = getClassNames(styles, {
       theme: theme!,
       className,
       disabled,
       label: !!label,
-      isDatePickerShown,
+      isDatePickerShown: isCalendarShown,
     });
 
-    const calloutId = getId('DatePicker-Callout');
-    const nativeProps = getNativeProps<React.HTMLAttributes<HTMLDivElement>>(this.props, divProperties, ['value']);
+    const nativeProps = getNativeProps<React.HTMLAttributes<HTMLDivElement>>(props, divProperties, ['value']);
     const iconProps = textFieldProps && textFieldProps.iconProps;
 
     return (
-      <div {...nativeProps} className={classNames.root}>
-        <div ref={this._datePickerDiv} aria-haspopup="true" aria-owns={isDatePickerShown ? calloutId : undefined}>
+      <div {...nativeProps} className={classNames.root} ref={forwardedRef}>
+        <div ref={datePickerDiv} aria-haspopup="true" aria-owns={isCalendarShown ? calloutId : undefined}>
           <TextField
             role="combobox"
             label={label}
-            aria-expanded={isDatePickerShown}
+            aria-expanded={isCalendarShown}
             ariaLabel={ariaLabel}
-            aria-controls={isDatePickerShown ? calloutId : undefined}
+            aria-controls={isCalendarShown ? calloutId : undefined}
             required={isRequired}
             disabled={disabled}
-            errorMessage={this._getErrorMessage()}
+            errorMessage={errorMessage}
             placeholder={placeholder}
             borderless={borderless}
             value={formattedDate}
-            componentRef={this._textField}
+            componentRef={textFieldRef}
             underlined={underlined}
             tabIndex={tabIndex}
             readOnly={!allowTextInput}
             {...textFieldProps}
-            id={this._id + '-label'}
+            id={id + '-label'}
             className={css(classNames.textField, textFieldProps && textFieldProps.className)}
             iconProps={{
               iconName: 'Calendar',
               ...iconProps,
               className: css(classNames.icon, iconProps && iconProps.className),
-              onClick: this._onIconClick,
+              onClick: onIconClick,
             }}
-            onKeyDown={this._onTextFieldKeyDown}
-            onFocus={this._onTextFieldFocus}
-            onBlur={this._onTextFieldBlur}
-            onClick={this._onTextFieldClick}
-            onChange={this._onTextFieldChanged}
+            // eslint-disable-next-line react/jsx-no-bind
```

Comment:
any reason to leave these functions as pure inline functions (and therefore need to disable the lint rule on them) rather than convert them to useCallbacks with empty dependency arrays?

Reply:
I'm been instructed by @xugao not to preemptively optimize with `useCallback` in Fabric code. 🤷‍♂️

Reply:
@lorejoh12 - if we need to keep these static in order to avoid re-rendering for the component that receives these, we can `useCallback`. but we don't want to over-memoize by `useCallback` everywhere.

Reply:
Sounds reasonable. Seems to me like we shouldn't have a lint rule suggesting that we use it in that case though.